juspay/hyperswitch
# feat(core): enable payment refund when payment is partially captured

## Type of Change



## Description

Currently, payment refund would only happen when intent status is `succeeded`(full capture).
With this change, it would be possible to refund the payment even when if is `partially captured`

### Additional Changes





## Motivation and Context



## How did you test it?

Manual
1. Create merchant with stripe connector.
2. Create a payment with confirm = true, capture = manual
3. Later, partially capture the payment ("amount_to_capture < amount"). (status changes to `partially_captured`)
4. Then do a refund on the above payment for the captured amount.

## Checklist
